### PR TITLE
docs(forms): add form documentation (#192)

### PR DESCRIPTION
**Pull request by mercs600**

### CODE REVIEW

File: docs/frontend/forms.md [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+      }
+    },
+  }
+}
+</script>
+```
+
+## Customize field templates
+
+Each form field use [`T3FormField`](https://github.com/TYPO3-Initiatives/nuxt-typo3/blob/master/lib/components/T3Form/T3FormField/T3FormField.vue) component as the base template. You can override this template:
```
Comment (SowaProgramuje) [inline docs/frontend/forms.md:119]: Each form field uses

File: docs/frontend/forms.md [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+  margin-right: 10px;
+}
+
+ul {
+  margin: 0;
+  color: red;
+}
+</style>
+```
+This component use [`<ValidationProvider/>`](https://vee-validate.logaretm.com/v3/api/validation-provider.html) for field validation.
```
Comment (SowaProgramuje) [inline docs/frontend/forms.md:179]: This component uses

File: docs/frontend/forms.md [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+This template will be suitable for most regular input elements. You can use this template to provide more form types like textarea, checkbox.
```
Comment (SowaProgramuje) [inline docs/frontend/forms.md:180]: You can use this template to provide more form elements like textarea, checkbox, ect.

File: docs/frontend/forms.md [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+                    "validators": [{ "identifier": "NotEmpty" }],
+                    "name": "tx_form_formframework[subject]"
+                  },
+                  
+                ]
+              },
+            ]
+```
+
+You can noticed that `name` and `subject` fields are nested in `Fieldset-2` which is nested in `Fieldset-1`. It means we have to render component fields in recursion.
```
Comment (SowaProgramuje) [inline docs/frontend/forms.md:243]: You may have noticed

File: docs/frontend/forms.md [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+                    "name": "tx_form_formframework[subject]"
+                  },
+                  
+                ]
+              },
+            ]
+```
+
+You can noticed that `name` and `subject` fields are nested in `Fieldset-2` which is nested in `Fieldset-1`. It means we have to render component fields in recursion. 
+For this list rendering is responsible [`T3FormFieldList` component](https://github.com/TYPO3-Initiatives/nuxt-typo3/blob/master/lib/components/T3Form/T3FormFieldList/T3FormFieldList.vue).
```
Comment (SowaProgramuje) [inline docs/frontend/forms.md:244]: [`T3FormFieldList` component] is responsible for rendering this list.

File: docs/frontend/forms.md [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+                  
+                ]
+              },
+            ]
+```
+
+You can noticed that `name` and `subject` fields are nested in `Fieldset-2` which is nested in `Fieldset-1`. It means we have to render component fields in recursion. 
+For this list rendering is responsible [`T3FormFieldList` component](https://github.com/TYPO3-Initiatives/nuxt-typo3/blob/master/lib/components/T3Form/T3FormFieldList/T3FormFieldList.vue). 
+
+This part of the template is responsible for matching frontend component with type or `field.identifier`.
```
Comment (SowaProgramuje) [inline docs/frontend/forms.md:246]: [with type] -> Is it `field type`?

File: docs/frontend/forms.md [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+
+For model validation we use `vee-validate`. 
+By default we have setup following rules:
+
+1. required as **NotEmpty**
+2. email as **EmailAddress**
+3. regex as **RegularExpression**
+
+
+All available rules you can find [here](https://vee-validate.logaretm.com/v3/guide/rules.html#rules), but remember to use the same rules name as in TYPO3.
```
Comment (SowaProgramuje) [inline docs/frontend/forms.md:323]: All additional rules are available [here]. Remember to use the same rules name as in TYPO3.

**Review by SowaProgramuje — CHANGES_REQUESTED** [review]

File: docs/frontend/forms.md [hunk 8]
```diff
@@ -0,0 +1,326 @@
+# Form handling
+
+TYPO3 Headless supports [Form Framework](https://docs.typo3.org/m/typo3/tutorial-editors/master/en-us/ContentElements/ContactForm/Index.html) and nuxt-typo3 supports form handling.
+
+### TYPO3 Headless Limitations
+At this moment we should be aware of some limitations on API level:
+1. Supports only one step forms (but still, you are able to build multistep forms on frontend side).
```
[the remark this answers is not in the document]
Comment (mercs600) [reply docs/frontend/forms.md:7]: there is no link at this moment, but we will add guide soon

File: docs/frontend/forms.md [hunk 9]
```diff
@@ -0,0 +1,326 @@
...
+You can use [T3Form component](https://github.com/TYPO3-Initiatives/nuxt-typo3/blob/master/lib/components/T3Form/T3Form.vue) for more specific cases, not only T3FormFramework. You can also override whole T3Form component to implement complicated scenarios, but before that just read the documentaion below, because we have kept a lot of ways to customization. 
+
+For model validation we have used [VeeValidate](https://vee-validate.logaretm.com/v3) plugin which in our opinion is the best way to validate forms in Vue.js applications. 
+
+## FormFramework Content Element
+Entry point for form handling is [`T3CeFormFormframework`](https://github.com/TYPO3-Initiatives/nuxt-typo3/blob/master/lib/components/T3CeFormFormframework/T3CeFormFormframework.vue) and this component is responsible for wraping `T3Form` and exchaning information between them. At this level you can override form template or default I18n labels.
```
[the remark this answers is not in the document]
Comment (mercs600) [reply docs/frontend/forms.md:21]: this is about internal i18n labels of this component

**Pull request merged**